storage/innobase/log/log0online.cc

```diff
@@ -1172,7 +1172,7 @@ log_online_write_bitmap_page(
 	}
 
 #ifdef UNIV_LINUX
-	posix_fadvise(log_bmp_sys->out.file, log_bmp_sys->out.offset,
+	posix_fadvise(log_bmp_sys->out.file.m_file, log_bmp_sys->out.offset,
```

Comment:
The abstraction level is off, please create a pfs_os_file-taking function in os0file.h. Probably OK to leave abstraction level violation in keeping POSIX_FADV_ macros at the callers though. #ifdef UNIV_LINUX should go inside the new function implementation